os/kernel/preference/preference_write.c

```diff
@@ -29,32 +29,28 @@
 #include <sys/stat.h>
 #include <tinyara/preference.h>
 
-#ifdef CONFIG_APP_BINARY_SEPARATION
+#if CONFIG_TASK_NAME_SIZE > 0
 #include "sched/sched.h"
```

Comment:
Please include `tinyara/sched.h` for `struct tcb_s`.

Reply:
"sched/sched.h" is needed for this_task(). I added <tinyara/sched.h> also.

Reply:
Yes. This is what I mean. Thanks.